spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DefaultEnvironmentPostProcessor.java

```diff
@@ -35,39 +35,53 @@
 import org.springframework.core.io.Resource;
 
 /**
- * Contributes the values from {@code dataflow-server.yml} if it exists, before any of Spring Boot's normal
+ * Contributes the values from {@code META-INF/dataflow-server-defaults.yml} and
+ * {@code dataflow-server.yml} if it exists, before any of Spring Boot's normal
  * configuration contributions apply. This has the effect of supplying overridable defaults to the
  * various Spring Cloud Data Flow Deployer SPI implementations that in turn override the defaults
  * provided by Spring Boot.
  *
  * @author Josh Long
+ * @author Janne Valkealahti
  */
 public class DefaultEnvironmentPostProcessor implements EnvironmentPostProcessor, Ordered {
 
-	private Resource resource = new ClassPathResource("/dataflow-server.yml");
+	private static Log logger = LogFactory.getLog(DefaultEnvironmentPostProcessor.class);
+	private final Resource serverResource = new ClassPathResource("/dataflow-server.yml");
+	private final Resource serverDefaultsResource = new ClassPathResource("META-INF/dataflow-server-defaults.yml");
```

Comment:
ah, I thought it was to remove `@EnableAutoConfiguration`   An  `@ImportAutoConfiguration` with just an exclude would maybe achieve the same thing as using the dataflow-server-defaults.yml file.